Auto build and create bumpup pr

kokeshiM0chi opened the pull request:
# Overview
This pr resolves this issue.

https://github.com/Finschia/finschia-wasm/issues/7

kokeshiM0chi committed:
fix: delete a file for osmosis

kokeshiM0chi committed:
feature: add command for `git checkout the tag` and call it

kokeshiM0chi committed:
feat: add rust action cache

kokeshiM0chi committed:
feat: add action to bump up proto files, build these and create this bump up pr automatically

kokeshiM0chi committed:
feat: add schedule trigger once a day

kokeshiM0chi committed:
feat: add linter and tests for pull request

kokeshiM0chi committed:
Merge remote-tracking branch 'upstream/main' into auto-build-and-create-bumpup-pr

170210 commented on .github/workflows/update-and-rebuild.yaml:
```diff
@@ -1,60 +1,106 @@
+          echo "CURRENT_TAG=$CURRENT_TAG" >> $GITHUB_ENV
+
+      - name: Check out to finschia
+        uses: actions/checkout@master
```
Please make it consistent with the other places.

170210 commented on .github/workflows/update-and-rebuild.yaml:
```diff
@@ -1,60 +1,106 @@
 
-      - name: Checkout
+      - name: Check out to finschia-wasm
+        if: env.NEW_TAG
         uses: actions/checkout@v3
```
use v4

170210 commented on .github/workflows/lint.yaml:
```diff
@@ -0,0 +1,26 @@
+  clippy-and-fmt:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
v4

170210 commented on .github/actions/cache/action.yaml:
```diff
@@ -0,0 +1,18 @@
+
+  steps:
+    - name: Cache cargo registry
+      uses: actions/cache@v2
```
v3

170210 commented on .github/workflows/lint.yaml:
```diff
@@ -0,0 +1,26 @@
+        if: success() || failure()
+        run: |
+          # `finschia-std` is excluded because of https://github.com/Finschia/finschia-wasm/issues/39
+          cargo clippy --workspace --exclude finschia-std --all-targets --all-features -- -D warnings
```
please add line after

170210 commented on packages/proto-build/src/main.rs:
```diff
@@ -45,17 +44,14 @@ pub fn generate(version_tags: &HashMap<String, String>) {
+    git::checkout_submodule(WASMD_DIR, &wasmd_version);
+    git::checkout_submodule(COMETBFT_DIR, &tendermint_version);
+    git::checkout_submodule(IBC_GO_DIR, &ibc_go_version);
+    git::checkout_submodule(ICS23_DIR, &ics23_version);
```
please check ics23_version. It seems been already deleted, but still in use.

kokeshiM0chi replied:
"ics23" will be supported once IBC support is available.
https://github.com/Finschia/finschia-wasm/issues/10

170210 reviewed (commented):
Why not separate the jobs, one for checking tags and the other for building and creating PRs.

kokeshiM0chi committed:
fix: separate the job to check whether the finschia tag is new and the job to update using the tag

kokeshiM0chi committed:
fix: upgrade the versions of actions/cache and actions/checkout

from actions/cache@v2 to actions/cache@v3
from actions/chackout@v3 to actions/checkout@v4

kokeshiM0chi committed:
fix: add line at end of file line

kokeshiM0chi commented:
@170210 

Thanks! I fixed it like this.
>one for checking tags and the other for building and creating PRs.

170210 commented on .github/workflows/update-and-rebuild.yaml:
```diff
@@ -1,60 +1,124 @@
+          echo "TENDERMINT_VERSION=$TENDERMINT_VERSION" >> $GITHUB_OUTPUT
+          echo "IBC_GO_VERSION=$IBC_GO_VERSION" >> $GITHUB_OUTPUT
+
+  create-bump-up-PR:
+    runs-on: ubuntu-latest
+    needs: check-new-finschia-tag
```
This job only works when `env.NEW_TAG` is not null, so you can add an `if` condition to the job, and this can reduce the number of if conditions in each step.

kokeshiM0chi replied:
This action is also used to modify the "packages/proto-build" code, so it is not limited to NEW_TAG.

kokeshiM0chi replied:
Sorry. I misunderstood. OK. I fixed here.
[fix: Change to perform "if: needs.check-new-finschia-tag.outputs.NEW_…](https://github.com/Finschia/finschia-wasm/pull/9/commits/b096dbcef769619315c784b00cf47e5136bc7c38)

kokeshiM0chi replied:
fixed.
[fix: swap the order of "job dependency definition" and "job dependenc…](https://github.com/Finschia/finschia-wasm/pull/9/commits/927e88ca2ab0d9894b99ee44e4f0c5fb496c85ae)

kokeshiM0chi committed:
fix: upgrade from actions/checkout@v3 to actions/checkout@v4

kokeshiM0chi committed:
fix: Change to perform "if: needs.check-new-finschia-tag.outputs.NEW_TAG" in job instead of checking "if: env.NEW_TAG" in each step

kokeshiM0chi committed:
fix: swap the order of "job dependency definition" and "job dependency source value condition check"

loloicci commented on .github/workflows/update-and-rebuild.yaml:
```diff
@@ -1,60 +1,116 @@
-          ./scripts/update-and-rebuild.sh "$OSMOSIS_REV"
+      - name: set each submodule version to env file in finschia-wasm
+        run: |
+          echo '# Choose from https://github.com/Finschia/finschia/releases' | cat > env
```
isn't it enough to use `> env` instead of `| cat > env`?

kokeshiM0chi replied:
Thanks. Fixed here.
[fix: delete unnecessary "| cat" when the env file is overwritten](https://github.com/Finschia/finschia-wasm/pull/9/commits/7895f81bdc181a0443311d3b4f338cb6be3b6702)
[fix: correct omission (delete unnecessary "| cat" when the env file i…](https://github.com/Finschia/finschia-wasm/pull/9/commits/211a3542f0af512f21ce6b62f65e4ee529adb6c8)

loloicci commented on .github/workflows/update-and-rebuild.yaml:
```diff
@@ -1,60 +1,116 @@
+      - name: Create Pull Request
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: Automated bump up ${{ env.NEW_TAG }}
```
how about adding a tag (maybe `build: `) for semantic commit messages?

kokeshiM0chi replied:
Thanks. Fixed here.
[fix: add the tag "build:" to the commit message of the auto bumpup pr](https://github.com/Finschia/finschia-wasm/pull/9/commits/f439cf5eb432cd7197155922f1d8379e9e96797b)

kokeshiM0chi committed:
fix: add the tag "build:" to the commit message of the auto bumpup pr

kokeshiM0chi committed:
fix: delete unnecessary "| cat" when the env file is overwritten

kokeshiM0chi committed:
fix: correct omission (delete unnecessary "| cat" when the env file is overwritten)
